src/operator/contrib/corner_pooling.cu

```diff
@@ -0,0 +1,293 @@
+    using mshadow::red::limits::MinValue;
+    const int b = index / (channel * height);
+    const int c = index % (channel * height) / height;
+    const int h = index % (channel * height) % height;
+    DType max_val = MinValue<DType>();
+    in_data += ((b*channel + c)*height + h)*width;
+    out_data += ((b*channel + c)*height + h)*width;
+
+    for (int w = w_start; w != w_end; w += w_step) {
+      const int index = w;
+      max_val = max_val > in_data[index] ? max_val : in_data[index];
+      out_data[index] = max_val;
+    }
+  }
+}
+
+
+template<typename DType>
+__global__ void CornerPoolingBackwardLRKernel(const int count,
+        const DType *out_data, const int batch,
+        const int channel, const int height,
+        const int width, const DType *out_grad, DType *in_grad,
+        int w_step, int w_start, int w_end) {
+  for (int index = (blockIdx.x + blockIdx.y*gridDim.x)*blockDim.x + threadIdx.x;
+       index < count;
+       index += blockDim.x * gridDim.x * gridDim.y) {
+    const int b = index / (channel * height);
+    const int c = index % (channel * height) / height;
+    const int h = index % (channel * height) % height;
+
+    out_data += ((b*channel + c)*height + h)*width;
+    out_grad += ((b*channel + c)*height + h)*width;
+    in_grad += ((b*channel + c)*height + h)*width;
+
+    int max_w_idx = w_start;
+    for (int w = w_start; w != w_end; w += w_step) {
+      const int index = w;
+      if (out_data[index] != out_data[max_w_idx]) {
+        max_w_idx = index;
+      }
+      in_grad[max_w_idx] += out_grad[index];
+    }
+  }
+}
+
+
+
+template<typename DType>
+inline void corner_pool(mshadow::Stream<gpu> *s, const DType *in_data,
+        const TShape &ishape, const int corner_pooling_type,
+        OpReqType req_type, DType *out_data) {
+  using mshadow::red::limits::MinValue;
+  CHECK_EQ(req_type, kWriteTo)
+      << "Only support req=kWriteTo in pooling operations";
+  int height = ishape[2], width = ishape[3];
+  if (corner_pooling_type == 0 || corner_pooling_type == 1) {
+    // top or bottom
+    int h_end = 0, h_start = 0, h_step = 0;
+    if (corner_pooling_type == 0) {
+      h_step = -1;
+      h_start = height - 1;
+      h_end = -1;
+    } else {
+      h_step = +1;
+      h_start = 0;
+      h_end = height;
+    }
+    const int count = ishape[0] * ishape[1] * width;
+    const int gridSize = (count + mshadow::cuda::kMaxThreadsPerBlock - 1) /
+                         mshadow::cuda::kMaxThreadsPerBlock;
+    dim3 dimGrid(mshadow::cuda::kMaxGridDim,
+                 (gridSize + mshadow::cuda::kMaxGridDim - 1) /
+                  mshadow::cuda::kMaxGridDim);
+    dim3 dimBlock(mshadow::cuda::kMaxThreadsPerBlock);
+    mshadow::cuda::CheckLaunchParam(dimGrid,
+            dimBlock, "Corner Pooling Forward");
+    cudaStream_t stream = mshadow::Stream<gpu>::GetStream(s);
+    CornerPoolingForwardTBKernel<DType><<<  dimGrid, dimBlock, 0, stream>>>
```

Comment:
can we add launch bound guards similar to https://github.com/apache/incubator-mxnet/pull/13188